### PR TITLE
Coin or update

### PR DESCRIPTION
**Pull request by entriken**
This update allows for a generic treatment of the 'mipgap' property across solvers.

### CODE REVIEW

File: tests/test_opt_solvers.py [hunk 1]
```diff
@@ -641,7 +641,7 @@ def test_solvers_on_LPs(self):
             
             try:
                 solver.solve(problem)
-            except opt.opt_solver.OptSolverError_NotAvailable:
+            except Exception:
```
Comment (awig) [inline tests/test_opt_solvers.py:644]: General exceptions are usually discouraged. What is the reason for changing this be less specific?

File: build_lib.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#! /bin/sh
+#! /bin/sh -v
```
Comment (awig) [inline build_lib.sh:1]: I'm gonna remove the verbose from the commit. The developer should add these themselves when needed.

File: tests/test_opt_solvers.py (hunk 1 above)
Comment (entriken) [reply tests/test_opt_solvers.py:644]: So true. I forget.  It was probably meant to be temporary for debugging.  I suggest to reject the change.

File: build_lib.sh (hunk 2 above)
Comment (entriken) [reply build_lib.sh:1]: Yes.

**Comment by awig** [comment]
Mumps is not compiling for me. Not sure if something has changed in the new version in IPOPT - how it is built. Trying to debug it.

**Comment by entriken** [comment]
I am not sure of your compiler error, but the following change was made at coin-or in their latest version of the distribution.

See in 'build_lib.sh' that there are two versions of how mumps is included.  For Ipopt-3.12.8, you can still use a 'ThirdParty' subdirectory and download with 'get.Mumps'.  For Ipopt-3.13.2, the 'ThirdParty' subdirectory is no longer used.  I assumed a super directory called 'ThirdParty-Mumps'.  The latter code is commented out.

File: tests/test_opt_solvers.py (hunk 1 above)
Comment (awig) [reply tests/test_opt_solvers.py:644]: Okay. I'll change it. Thanks

**Comment by awig** [comment]
Thanks, might be a version problem with clang or gfortran.  Would you please let me know what you are on?
Here are mine:
```
> gfortran --version
GNU Fortran (Homebrew GCC 10.2.0) 10.2.0
...

> clang --version
Apple clang version 11.0.3 (clang-1103.0.32.62)
```

First error reached:
```
Making all in ThirdParty/Mumps
/bin/sh ./../../libtool --tag=F77 --mode=compile gfortran  -I. -I`echo .`  -I`echo ./MUMPS/src` -I`echo ./MUMPS/libseq` -I`echo ./MUMPS/include`   -O3 -pipe    -c -o tools_common_mod.lo `test -f 'MUMPS/src/tools_common_mod.F' || echo './'`MUMPS/src/tools_common_mod.F
mkdir .libs
 gfortran -I. -I. -I./MUMPS/src -I./MUMPS/libseq -I./MUMPS/include -O3 -pipe -c MUMPS/src/tools_common_mod.F  -fno-common -o .libs/tools_common_mod.o
/bin/sh ./../../libtool --tag=F77 --mode=compile gfortran  -I. -I`echo .`  -I`echo ./MUMPS/src` -I`echo ./MUMPS/libseq` -I`echo ./MUMPS/include`   -O3 -pipe    -c -o dmumps_comm_buffer.lo `test -f 'MUMPS/src/dmumps_comm_buffer.F' || echo './'`MUMPS/src/dmumps_comm_buffer.F
 gfortran -I. -I. -I./MUMPS/src -I./MUMPS/libseq -I./MUMPS/include -O3 -pipe -c MUMPS/src/dmumps_comm_buffer.F  -fno-common -o .libs/dmumps_comm_buffer.o
MUMPS/src/dmumps_comm_buffer.F:2667:23:

 2658 |         CALL MPI_PACK( WHAT, 1, MPI_INTEGER,
      |                       2
......
 2667 |         CALL MPI_PACK( LIST_SLAVES, NSLAVES, MPI_INTEGER,
      |                       1
Error: Rank mismatch between actual argument at (1) and actual argument at (2) (scalar and rank-1)
MUMPS/src/dmumps_comm_buffer.F:2670:23:
```

**Comment by entriken** [comment]
$ gfortran --version
GNU Fortran (Homebrew GCC 9.2.0_2) 9.2.0
$ clang --version
Apple clang version 12.0.0 (clang-1200.0.32.21)

The rank mismatch implies that there is a matrix versus scalar argument conflict.  This is new for me in fortran, but means there is may be a mismatch in the versioning between the defined function and the calling function.

**Comment by awig** [comment]
Might be an issue with gfortran version flags. Need to reinstall xcode to version 12.1 anways and then try to install older version of gcc via homebrew

**Comment by entriken** [comment]
I think the error message is pointing out that within one file there is a call to MPI_PACK using a scalar (INTEGER) argument, 'CALL MPI_PACK( WHAT', and elsewhere this is a call using an array argument (INTEGER(*)).  This may or not result in a runtime error.  I wonder what version of mumps you using?  Also, is there a strict dependency in mumps on using MPI, or is this optional?

**Comment by awig** [comment]
I believe mumps should be strictly tied to the version that IPOPT is getting with the .get.MUMPS script. I wouldn't want to modify it. I'm working on changing the version but reinstalling xcode takes a while.

**Comment by awig** [comment]
what a pain to figure out. I reinstalled xcode, xcode command line tools to version 12.1. Tried installing gcc from homebrew with different versions but that didn't work as the only one that seems to work properly is 10.3. uninstalled gcc and installed gfortran directly via homebrew with version 8.3 and it works. So, to summarize MUMPS 4.10 that is installed for Ipopt version 3.12.8 does not build with gfortran version 10.3. Need to make sure using version < 10. 

All tests are passing now. 

Thanks for upgrading the packages and extending the CBC parameters. I'll commit minor changes and merge.

**Pull request merged**